test/parallel/test-fs-promises-file-handle-close.js

```diff
@@ -0,0 +1,39 @@
+// Flags: --expose-gc --no-warnings
+'use strict';
+
+// Test that a runtime warning is emitted when a FileHandle object
+// is allowed to close on garbage collection. In this future, this
+// test should verify that closing on garbage collection throws a
+// process fatal exception.
+
+const common = require('../common');
+const assert = require('assert');
+const { promises: fs } = require('fs');
+
+const warning =
+  'Closing a FileHandle object on garbage collection is deprecated. ' +
+  'Please close FileHandle objects explicitly using ' +
+  'FileHandle.prototype.close(). In the future, an error will be ' +
+  'thrown if a file descriptor is closed during garbage collection.';
+
+async function doOpen() {
+  const fh = await fs.open(__filename);
+
+  common.expectWarning({
+    Warning: [[`Closing file descriptor ${fh.fd} on garbage collection`]],
+    DeprecationWarning: [[warning, 'DEP00XX']]
+  });
+
+  return fh;
+}
+
+// Perform the file open assignment within a block.
+// When the block scope exits, the file handle will
+// be eligible for garbage collection.
+{
+  doOpen().then(common.mustCall((fd) => {
+    assert(fd);
```

Comment:
Nit: Personally, I am not a fan of using `assert()` for anything other than booleans because it makes it harder for the reader to understand what the assertion is supposed to prevent: Does it prevent `fd === undefined`, `fd === false`, `fd === null`, or `fd === 0`? If all of these, I would use `assert(fd > 0)`.

Reply:
Maybe this?:

```suggestion
assert.strictEqual(typeof fd, 'number');
```

Reply:
Or this?:

```suggestion
    assert.ok(fd > 2); // fd 2 is stderr; open should return 3 or higher
```